pom.xml

```diff
@@ -1136,6 +1148,7 @@
                                 <ignoredDependency>org.junit.platform</ignoredDependency>
                                 <ignoredUnusedDeclaredDependency>io.micrometer:micrometer-registry-prometheus</ignoredUnusedDeclaredDependency>
                                 <ignoredUnusedDeclaredDependency>org.mockito:mockito-inline</ignoredUnusedDeclaredDependency>
+                                <ignoredUnusedDeclaredDependency>io.javaoperatorsdk:jenvtest-fabric8-client-support</ignoredUnusedDeclaredDependency>
```

Comment:
I don't follow how you actually use it when it is not used. Can you elaborate on it?

Reply:
Yeah sure, it is only used to enable this injection of the fabric8 client but the dependency analyzer doesn't pick that up
```
// Injected by jenvtest Kubernetes Api Server
    @SuppressWarnings("unused")
    static KubernetesClient client;
```

Reply:
> So how does it differentiate from the other mock server?

Answering here to keep it in this thread/conversation.

The `@EnableKubeAPIServer` annotation on the test is setting up the jenvtest kubernetes api server. This gives an api server that actually supports SSA, so we are not using the mock client anymore which worked fine for crud operations but does not support SSA (the mock client is used everywhere else via`@EnableKubernetesMockClient`). `@EnableKubeAPIServer` comes from the `io.javaoperatorsdk:jenvtest` dependency.

Now that we are not using a mocked client, we still need a client to talk to the api server, and the `io.javaoperatorsdk:jenvtest-fabric8-client-support` dependency gives us this, allowing us to inject a `KubernetesClient` into the test.

The dependency analyzer doesn't know how this `KubernetesClient` is being injected and so flags the additional dependency as an `UnusedDeclaredDependency`.

Hope that answers the question, let me know if it doesn't.

Reply:
Got it. Thanks.